pkg/util/ftp/auth.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ftp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/crypto/ssh"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"antrea.io/antrea/pkg/apis/controlplane"
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+)
+
+const (
+	secretKeyWithAPIKey      = "apikey"
+	secretKeyWithBearerToken = "token"
+	secretKeyWithUsername    = "username"
+	secretKeyWithPassword    = "password"
+)
+
+// GenSSHClientConfig generates ssh.ClientConfig from username and password
+func GenSSHClientConfig(username, password string) *ssh.ClientConfig {
+	cfg := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{ssh.Password(password)},
+		// #nosec G106: skip host key check here and users can specify their own checks if needed
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+		Timeout:         time.Second,
+	}
+	return cfg
+}
+
+// ParseBundleAuth returns the authentication from the Secret provided in BundleServerAuthConfiguration.
+// The authentication is stored in the Secret Data with a key decided by the AuthType, and encoded using base64.
+func ParseBundleAuth(authentication crdv1alpha1.BundleServerAuthConfiguration, kubeClient clientset.Interface) (*controlplane.BundleServerAuthConfiguration, error) {
```

Comment:
this is meant to be a generic function now, so tying it to the support-bundle feature is not appropriate